Engine/src/Engine/InputSystem.cpp

```diff
@@ -0,0 +1,110 @@
+#include "Engine/InputSystem.h"
+#include "GLFW/glfw3.h"
+
+using namespace Engine;
+
+InputSystem* InputSystem::m_instance = nullptr;
+std::mutex InputSystem::m_instanceMutex;
+
+
+InputSystem* InputSystem::getInstance()
+{
+	std::unique_lock<std::mutex> lock;
+
+	if (!m_instance)

```

Comment:
Would this check for nullptr, or is it just checking for NULL? I recall nullptr being a guideline thing, but if null is better in this case, that's fair.

Reply:
This checks for both NULL and nullptr